### PR TITLE
[sonic-breakout_cfg.yang]: Remove pattern from sonic-breakout_cfg.yang.

### PR DESCRIPTION
**Pull request by praveen-li**
[sonic-breakout_cfg.yang]: Remove pattern from sonic-breakout_cfg.yang.
    
    Changes:
    -- Remove pattern from sonic-breakout_cfg.yang, it is redundant.
    -- test changes.

Signed-off-by: Praveen Chaudhary <pchaudhary@linkedin.com>

#### Why I did it

 -- Remove pattern from sonic-breakout_cfg.yang, it is redundant.
Breakout Mode is verified in Sonic-utilities using platform.json

#### How I did it

Remove pattern from sonic-breakout_cfg.yang
.
#### How to verify it

Built the packages after changing the tests.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012

### CODE REVIEW

**Comment by kwangsuk** [comment]
There is a reference to  BREAKOUT_CFG table and yang.
https://github.com/Azure/SONiC/blob/master/doc/dynamic-port-breakout/sonic-dynamic-port-breakout-HLD.md

Don't you need to update the HLD accordingly?

**Comment by praveen-li** [comment]
Hi Kwan

There is no reference in HLD for breakout_cfg yang models. It's me know if you have seen it.
Note that the breakout table will still exist we are just removing the Yang model corresponding to it,

Thanks,

Regards
Praveen
________________________________

**Comment by kwangsuk** [comment]
@praveen-li 
Oh, ok. So, the BREAKOUT_CFG table will still exist. Then any reason to remove YANG file?

**Comment by praveen-li** [comment]
> @praveen-li
> Oh, ok. So, the BREAKOUT_CFG table will still exist. Then any reason to remove YANG file?

Possible breakout modes are controlled by platform.json and are thoroughly checked by the Python code. So we do not need extra layer of checking using Yang model because then we have to maintain it at 2 places.

Also breakout CFG table doesn't impact backend.

**Comment by praveen-li** [comment]
/Azure.sonic-buildimage run

**Comment by anshuv-mfst** [comment]
3/11: PR reviewed in YANG subgroup. 
@zhenggen-xu - could you please review as well.

**Comment by lguohan** [comment]
@praveen-li, who to sign off?

**Comment by praveen-li** [comment]
> @praveen-li, who to sign off?

@lguohan I think Xu can review this one, once I resolve the conflicts.
@zhenggen-xu

**Review by Prasanth-KV — DISMISSED** [review]

**Comment by zhenggen-xu** [comment]
After some more thought, I think we don't want to treat this table BREAKOUT_CFG very differently.  The main problem for this table with Yang models is that the breakout mode is not known for all platforms and hard to maintain.  Maybe we can relax that in the Yang model file, and we validate that by the sonic-utilities or other plugins (which was the case already).  The yang model here is still good to have at least to validate the sanity of field name etc. This also makes the table to be consistent with other tables.

**Review by zhenggen-xu — CHANGES_REQUESTED** [review]
As requested

**Comment by praveen-li** [comment]
> After some more thought, I think we don't want to treat this table BREAKOUT_CFG very differently. The main problem for this table with Yang models is that the breakout mode is not known for all platforms and hard to maintain. Maybe we can relax that in the Yang model file, and we validate that by the sonic-utilities or other plugins (which was the case already). The yang model here is still good to have at least to validate the sanity of field name etc. This also makes the table to be consistent with other tables.

@zhenggen-xu : I guess then , it makes sense to go ahead by just removing the pattern in this yang model and we can still keep the yang model file. 
```
                leaf brkout_mode {
                    type string {
                        length 1..64;
                        /*
                         * Below allowed patterns are based on most used Platforms,
                         * Add any other breakout-mode to allow Dynamic Port
                         * Breakout to that breakout-mode.
                         */
                        pattern '1x100G\[40G\]|2x50G|4x25G\[10G\]|2x25G\(2\)\+1x50G\(2\)|1x50G\(2\)\+2x25G\(2\)|1x400G|2x200G|4x100G|8x50G|1x200G\[100G,50G,40G,25G,10G,1G\]|2x100G\[50G,40G,25G,10G,1G\]|4x50G\[40G,25G,10G,1G\]|1x25G\[10G\]|1x100G\[50G,40G,25G,10G\]|2x50G\[40G,25G,10G\]|4x25G\[10G\]|1x25G\[10G,1G\]|1x100G\[50G,40G,25G,10G,1G\]|2x50G\[40G,25G,10G,1G\]|4x25G\[10G,1G\]|1x400G\[200G,100G,50G,40G,25G,10G,1G\]|2x200G\[100G,50G,40G,25G,10G,1G\]|4x100G\[50G,40G,25G,10G,1G\]';
                    }
                }
```

**Comment by zhenggen-xu** [comment]
> > After some more thought, I think we don't want to treat this table BREAKOUT_CFG very differently. The main problem for this table with Yang models is that the breakout mode is not known for all platforms and hard to maintain. Maybe we can relax that in the Yang model file, and we validate that by the sonic-utilities or other plugins (which was the case already). The yang model here is still good to have at least to validate the sanity of field name etc. This also makes the table to be consistent with other tables.
> 
> @zhenggen-xu : I guess then , it makes sense to go ahead by just removing the pattern in this yang model and we can still keep the yang model file.
> 
> ```
>                 leaf brkout_mode {
>                     type string {
>                         length 1..64;
>                         /*
>                          * Below allowed patterns are based on most used Platforms,
>                          * Add any other breakout-mode to allow Dynamic Port
>                          * Breakout to that breakout-mode.
>                          */
>                         pattern '1x100G\[40G\]|2x50G|4x25G\[10G\]|2x25G\(2\)\+1x50G\(2\)|1x50G\(2\)\+2x25G\(2\)|1x400G|2x200G|4x100G|8x50G|1x200G\[100G,50G,40G,25G,10G,1G\]|2x100G\[50G,40G,25G,10G,1G\]|4x50G\[40G,25G,10G,1G\]|1x25G\[10G\]|1x100G\[50G,40G,25G,10G\]|2x50G\[40G,25G,10G\]|4x25G\[10G\]|1x25G\[10G,1G\]|1x100G\[50G,40G,25G,10G,1G\]|2x50G\[40G,25G,10G,1G\]|4x25G\[10G,1G\]|1x400G\[200G,100G,50G,40G,25G,10G,1G\]|2x200G\[100G,50G,40G,25G,10G,1G\]|4x100G\[50G,40G,25G,10G,1G\]';
>                     }
>                 }
> ```

Yes, let's keep the file, but not maintain the pattern here, the pattern validation is done by platform specific platform.json and library.

**Comment by anshuv-mfst** [comment]
@praveen-li - can you please address review comments and close on open items.

File: src/sonic-yang-models/yang-models/sonic-breakout_cfg.yang [hunk 1]
```diff
@@ -40,14 +41,9 @@ module sonic-breakout_cfg {
                 }
 
                 leaf brkout_mode {
+                    description "break_mode for port, it is verifed using platform.json";
```
Comment (venkatmahalingam) [inline src/sonic-yang-models/yang-models/sonic-breakout_cfg.yang:44]: Please fix the typo.

**Comment by zhangyanzhao** [comment]
@zhenggen-xu  will take a second approval, then merge by Guohan.

**Review by dgsudharsan — APPROVED** [review]

**Comment by zhangyanzhao** [comment]
@lguohan can you please merge this PR? Thanks.

**Review by zhenggen-xu — APPROVED** [review]

**Comment by zhangyanzhao** [comment]
will merge after the test passed

**Comment by zhangyanzhao** [comment]
azpw /run

**Review by lguohan — APPROVED** [review]

**Pull request merged**